Mathlib/RingTheory/Localization/BaseChange.lean

```diff
@@ -0,0 +1,46 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang, Jujian Zhang
+-/
+import Mathlib.RingTheory.IsTensorProduct
+import Mathlib.RingTheory.Localization.Module
+
+/-!
+# Localized Module
+
+Given a commutative ring `R`, a multiplicative subset `S ⊆ R` and an `R`-module `M`, we can
+localize `M` by `S`. This gives us a `Localization S`-module.
+
+## Main definitions
+
+* `IsLocalizedModule.IsBaseChange` : A localization of modules is a base change.
+-/
+
+variable {R : Type*} [CommRing R] (S : Submonoid R)
+  (A : Type*) [CommRing A] [Algebra R A] [IsLocalization S A]
+  {M : Type*} [AddCommMonoid M] [Module R M] [Module A M] [IsScalarTower R A M]
+  {M' : Type*} [AddCommMonoid M'] [Module R M'] [Module A M'] [IsScalarTower R A M']
+  (f : M →ₗ[R] M') [IsLocalizedModule S f]
+
+/-- If `(f : M →ₗ[R] M')` is a localization of modules, then the map
+`(localization S) × M → N, (s, m) ↦ s • f m` is the tensor product (insomuch as it is the universal
+bilinear map).
+In particular, there is an isomorphism between `LocalizedModule S M` and `(Localization S) ⊗[R] M`
+given by `m/s ↦ (1/s) ⊗ₜ m`.
+-/
+theorem IsLocalizedModule.isBaseChange : IsBaseChange A f := by
```

Comment:
I think this should be `iff` too ...

Reply:
The reverse direction can be derived from the current direction using [IsBaseChange.equiv](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/IsTensorProduct.html#IsBaseChange.equiv) and the missing lemma that `IsLocalizedModule` is preserved under composition with LinearEquiv.

Reply:
Yes, that's clever. I pushed the new version. Thanks!

Reply:
I tried and found it hard to directly show the tensor product is the localized module. Then I recall there were [some Zulip discussions](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/localization.20map.20for.20modules/near/275122603) that this is likely not possible without an explicit construction of the localized module. Utilizing the construction `LocalizedModule` is exactly what we do in this clever proof.